app/pb_kits/playbook/pb_layout/layout.rb

```diff
@@ -2,92 +2,44 @@
-                     full: default_configuration,
-                     id: default_configuration,
-                     position: default_configuration,
-                     transparent: default_configuration,
-                     size: default_configuration,
-                     &block)
-        self.configured_classname = classname
-        self.configured_collapse = collapse
-        self.configured_dark = dark
-        self.configured_data = data
-        self.configured_full = full
-        self.configured_id = id
-        self.configured_position = position
-        self.configured_transparent = transparent
-        self.configured_size = size
-        self.block = block_given? ? block : nil
-      end
-
-      def position
-        if configured_position == default_configuration
-          "_left"
-        else
-          "_#{configured_position}"
-        end
-      end
-
-      def transparent
-        true_value(configured_transparent, "_transparent", "")
-      end
+      partial "pb_layout/layout"
 
-      def size
-        if configured_size == default_configuration
-          "md"
-        else
-          configured_size
-        end
+      prop :collapse, type: Playbook::Props::Enum,
+                      values: %w[xs sm md lg xl],
+                      default: "xs"
+      prop :dark, type: Playbook::Props::Boolean, default: false
+      prop :full, type: Playbook::Props::Boolean, default: false
+      prop :position, type: Playbook::Props::Enum,
+                      values: %w[left right],
+                      default: "left"
+      prop :transparent, type: Playbook::Props::Boolean, default: false
+      prop :size, type: Playbook::Props::Enum,
+                      values: %w[xs sm md lg xl],
+                      default: "md"
+      def dark_class
+        dark ? "dark" : nil
       end
 
-      def collapse
-        if configured_collapse == default_configuration
-          " layout#{position}_collapse_xs"
-        else
-          " layout#{position}_collapse_#{configured_collapse}"
-        end
+      def full_class
+        full ? "full" : ""
       end
 
-      def full
-        true_value(configured_full, " full", "")
+      def transparent_class
+        transparent ? "transparent" : nil
       end
 
-      def dark
-        true_value(configured_dark, "_dark", "")
+      def collapse_class
+        "layout_#{position}_collapse_#{collapse}"
       end
 
-      def yield(context:)
-        context.capture(&block)
-      end
-
-      def to_partial_path
-        "pb_layout/layout"
-      end
-
-    private
-
-      DEFAULT = Object.new
-      private_constant :DEFAULT
-      def default_configuration
-        DEFAULT
+      def classname
+        first_class = generate_classname("pb_layout", size, position, dark_class, transparent_class)
+        classname = first_class + full_class + collapse_class
+        [first_class, full_class, collapse_class].reject(&:empty?).join(" ")
```

Comment:
Since all of the `_class` methods are now only used by the `classname` method, they can all be moved into `private` access.